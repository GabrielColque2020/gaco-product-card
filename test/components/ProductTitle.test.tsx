import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ProductTitle, ProductCard } from "../../src/";
import { product1 } from "../data/products";


describe('ProductTitle', () => {
	test('debe de mostrar el componente correctamente con el titulo personalizado', () => {
		const rerender = render(<ProductTitle title={"Custom Product"} className={"custom-class"} />);
		expect(rerender.baseElement).toMatchSnapshot();
	});

	test('debe de mostrar el componente con el nombre del producto', () => {
		const rerender = render(
			<ProductCard product={product1}>
				{
					()=> (
						<ProductTitle/>
					)
				}
			</ProductCard>
		);
		expect(rerender.baseElement).toMatchSnapshot();
	});
});