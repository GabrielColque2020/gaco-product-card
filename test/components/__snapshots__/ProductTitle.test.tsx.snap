// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<body>
  <div>
    <div
      class="productCard undefined"
    >
      <span
        class="productDescription undefined"
      >
        Coffee Mug - Sin Imagen
      </span>
    </div>
  </div>
</body>
`;

exports[`ProductTitle debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<body>
  <div>
    <span
      class="productDescription custom-class"
    >
      Custom Product
    </span>
  </div>
</body>
`;
